packages/jsx-email/src/renderer/raw.ts

```diff
@@ -1,12 +1,94 @@
 const START_TAG = '__COMMENT_START';
 const END_TAG = '__COMMENT_END';
+
+// Note: opaque wrapper tokens used to safely carry raw payloads through
+// rehype/HTML parsing without introducing nested HTML comments.
+export const RAW_WRAPPER_START = '__JSX_EMAIL_RAW_START__';
+export const RAW_WRAPPER_END = '__JSX_EMAIL_RAW_END__';
+
+const escapeRe = (s: string) => s.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+
+// Exported regular expressions for Raw wrapper handling. Keeping these as
+// named exports makes their behavior explicit and testable.
+export const RAW_MARKER_CONTENT_RE = new RegExp(
+  `${escapeRe(RAW_WRAPPER_START)}([\\s\\S]*?)${escapeRe(RAW_WRAPPER_END)}`
+);
+
+// Allow an optional stray "--" to precede the closing angle bracket in the
+// closing tag. Some HTML serializers can emit a comment terminator sequence
```

Comment:
I don't understand the accounting for "some html serializers" when we have exactly one in this project. shouldn't the code be fixed so that stray "--" are not emitted?